src/jdk.compiler/share/classes/com/sun/tools/javac/file/BaseFileManager.java

```diff
@@ -191,7 +193,9 @@ public void clear() {
     }
 
     protected ClassLoader getClassLoader(URL[] urls) {
-        ClassLoader thisClassLoader = getClass().getClassLoader();
+        ClassLoader thisClassLoader = CodeReflectionSupport.CODE_LAYER != null ?
```

Comment:
We need to use a "base" classloader which is the same as the one used to build the `jdk.incubator.code` module layer (if it exists), otherwise we won't be able to load types defined there.